shared/src/reentrancy_guard.rs

```
@@ -0,0 +1,47 @@
+#![no_std]
```

Clippy (1.95.0)

[warning] the `#![no_std]` attribute can only be used at the crate root

(warning)